heliopy/data/cluster.py

```diff
@@ -75,7 +78,10 @@ def processing_func(file):
             if 'CDF_EPOCH' in file.varget(key, expand=True).values():
                 index_key = key
                 break
-        return util.cdf2df(file, index_key)
+
+        return util.cdf2xr(file, starttime, endtime, index_key, product_list)
```

Comment:
@dstansby I think here is a point open for discussion. We identified several points:
- using xarray for both distributions and also timeseries (1D) could help by providing a unique format for the data, avoiding the check for distribution function/timeseries
- on the other hand, what would be the potential advantages of using pandas for timeseries ?
- only from the point of view of our project, we plan to use a single data container (xarray), what are your plans concerning HelioPy?

What is your opinion on each point?

Reply:
I agree that it would be ideal to have a similar format in both cases, but HelioPy is now at the point where quite a few people use it, so we should maintain backwards compatibility in order to not break other people's code (who are relying on 1D data being a `TimeSeries` obejct which has pandas stuff underneath).

I think a good compromise might be to add an example showing how to convert a `TimeSeries` object to an `xarray` for those who want to do that, which I would be happy to write.

Reply:
I fully understand. What about an optional argument like 'xarray'. If no argument is passed we stick to the check for 1D timeseries to return pandas df and keep the exact same signature. If the argument is passed, it returns an xarray.

Reply:
That sounds 👍  to me -  it could just be a keyword argument that defaults to `False`.

Reply:
indeed, I think this is the best solution, and the first I imagined so I should still have a commit with a keyword something like want_ts=True. IMO, to include this modification the easiest way is to load everything in xarray as it is, and by default the xarray is transformed to a DataFrame using the .to_dataframe() included in xarray, then construct the SunPy TimeSeries. What do you think ?

Reply:
Yep, that sounds like a good way to do it